src/scripts/weather/WeatherProvider.js

```diff
@@ -0,0 +1,41 @@
+import keyObj from "../Settings.js"
+
+const eventHub = document.querySelector(".container")
+
+//will use once event listener is created.
+// let eventPostalCode
+
+// const eventHub = document.querySelector(".container")
+// eventHub.addEventListener("eventSelected", customEvent => {
+//     eventPostalCode = customEvent.detail.zip.substring(0, 5)
+// })
+
+let weather = []
+
+eventHub.addEventListener("showEventWeather", (event) => {
+    event.target.detail === zipCode
+})
```

Comment:
I don't recommend having event listeners inside the provider. Instead, I would recommend calling getWeather and useWeather from weatherList. Just trying to keep consistency. 😃